x-pack/plugins/session_view/public/components/ProcessTree/helpers.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Process, ProcessEvent, ProcessMap } from '../../../common/types/process_tree';
+import { ProcessImpl } from './hooks';
+
+export const updateProcessMap = (processMap: ProcessMap, events: ProcessEvent[]) => {
+  events.forEach((event) => {
+    const { entity_id: id } = event.process;
+    let process = processMap[id];
+
+    if (!process) {
+      process = new ProcessImpl(id);
+      processMap[id] = process;
+    }
+
+    process.events.push(event);
+  });
+
+  return processMap;
+};
+
+export const buildProcessTree = (
+  processMap: ProcessMap,
+  events: ProcessEvent[],
+  orphans: Process[],
+  sessionEntityId: string,
+  backwardDirection: boolean = false
+) => {
+  if (backwardDirection) {
+    events = events.slice().reverse();
```

Comment:
Nit: Potentially cache this in another value, like `sortedEvent` to not mutate the original arg.